File: Programming Language(ver.2)/C#/503.cs
using System;

public class TQC503
{
    public static void Main()
    {
        string input_a = Console.ReadLine();
        string input_b = Console.ReadLine();
        int a = Convert.ToInt32(input_a);
        int b = Convert.ToInt32(input_b);
        double result_double;
        for (double i = 1; i < a; i++)
        {
            result_double = Math.Sqrt(i);
            if (Convert.ToInt32(result_double) == result_double)
            {
                result_double = Math.Pow(result_double, b);
                Console.WriteLine(result_double);
            }
        }
    }
}

File: Programming Language(ver.2)/Go/503.go
package main
import (
	"fmt"
	"math"
)
func main(){
	var a, b int
	fmt.Scan(&a)
	fmt.Scan(&b)
	for i := 1; i < a; i++ {
		sqrt_a := math.Sqrt(float64(i))
		if sqrt_a != math.Trunc(sqrt_a) {
			continue;
		}
		fmt.Printf("%.0f\n", math.Pow(sqrt_a, float64(b)))
	}
}
